superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx

```diff
@@ -244,9 +244,23 @@ class DatasourceControl extends React.PureComponent {
     return (
       <Styles data-test="datasource-control" className="DatasourceControl">
         <div className="data-container">
-          <Icons.DatasetPhysical className="dataset-svg" />
-          {/* Add a tooltip only for long dataset names */}
-          {!isMissingDatasource && datasource.name.length > 25 ? (
+          {isValidDatasourceType ? (
+            <Icons.ConsoleSqlOutlined className="datasource-svg" />
+          ) : (
+            <Icons.DatasetPhysical className="datasource-svg" />
+          )}
+          {isValidDatasourceType ? (
+            /* Add a tooltip only for long dataset names */
+            !isMissingDatasource && datasource.sql.length > 25 ? (
```

Comment:
curious here why datasource.sql.length have to be greater than 25?

Reply:
This was how it was initially set, I just added the query route. I'm also not sure why 25 haha

Reply:
can we make this value a constant then?